product_variant_sale_price/models/product_product.py

```diff
@@ -0,0 +1,107 @@
+# Copyright 2016 Sergio Teruel <sergio.teruel@tecnativa.com>
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+
+from odoo import api, fields, models
+
+
+class ProductTemplate(models.Model):
+    _inherit = "product.template"
+
+    def _prepare_variant_values(self, combination):
+        values = super()._prepare_variant_values(combination)
+        values["fix_price"] = self.list_price
+        return values
+
+    def _create_variant_ids(self):
+        res = super()._create_variant_ids()
+        for tmpl in self:
+            for variant in tmpl.product_variant_ids:
+                variant.with_context(skip_update_fix_price=True).write(
+                    {"fix_price": tmpl.list_price}
+                )
+        return res
+
+    def write(self, vals):
+        res = super().write(vals)
+        if self.env.context.get("skip_update_fix_price", False):
+            return res
+        if "list_price" in vals:
+            self.mapped("product_variant_ids").write({"fix_price": vals["list_price"]})
+        return res
+
+    def _get_combination_info(
+        self,
+        combination=False,
+        product_id=False,
+        add_qty=1,
+        parent_combination=False,
+        only_template=False,
+    ):
+        res = super()._get_combination_info(
+            combination,
+            product_id,
+            add_qty,
+            parent_combination,
+            only_template,
+        )
+        res["price_extra"] = 0.0
+        return res
+
+
+class ProductProduct(models.Model):
```

Comment:
@rousseldenis Hi Denis, i didn't found an answer in the [contribution guide](https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/CONTRIBUTING.rst#modules). But should the two classes be split into two files?

Reply:
As you wish, IMO I prefer split into 2 files. But there is not standard rule for that